doc/ref/gappkg.xml

```diff
@@ -173,5 +184,2074 @@ users.
+
+Package files may follow the style used for the &GAP; library.
+Every file in the &GAP; library starts with a header that lists the 
+filename, copyright, a short description of the file contents and the 
+original authors of this file, and ends with a comment line <C>#E</C>. 
+Indentation in functions and the use of decorative spaces in the code 
+are left to the decision of the authors of each file. Global (i.e. 
+re-used elsewhere) comments usually are indented by two hash marks and 
+two blanks, in particular, every declaration or method or function 
+installation which is not only of local scope is separated by a header.
+<P/>
+Historically, when the &GAP; main manuals were based on the &TeX; macros 
+described in the document <Q>The gapmacro.tex Manual Format</Q>
+(the file <F>gap4rN/doc/gapmacrodoc.pdf</F>)
+such headers were used for the manuals and have the type
+<P/>
+<Log><![CDATA[
+#############################################################################
+##
+#X  ExampleFunction(<A>,<B>)
+##
+##  This function does great things.
+]]></Log>
+<P/>
+where <Q>X</Q> was one of the letters: <C>F</C>, <C>A</C>, <C>P</C>, <C>O</C>,
+<C>C</C>, <C>R</C> or <C>V</C> indicating whether the object declared will
+be a function, attribute, property, operation, category, representation or 
+variable, respectively. Additionally <C>M</C> was used in <C>.gi</C> files 
+for method installations. Then a sample usage of the function was given,
+followed by a comment which described the identifier.
+This description was automatically be extracted to build the
+manual source, if there is a <C>\Declaration</C> line in some
+<C>.msk</C> file together with an appropriate configuration file.
+<P/>
+Nowadays, facilities to distribute a document over several files
+to allow the documentation for parts of some code to be stored in
+the same file as the code itself are provided by the &GAPDoc; package (see
+<Ref Sect="Distributing a Document into Several Files" BookName="gapdoc"/>).
+The same approach is demonstrated by the <Package>Example</Package> package.
+E.g. <F>example/doc/example.xml</F> has the statement 
+<C>&lt;#Include Label="ListDirectory"></C>
+and <F>example/lib/files.gd</F> contains
+<Log><![CDATA[
+#############################################################################
+##
+#F  ListDirectory([<dir>])  . . . . . . . . . . list the files in a directory
+##
+##  <#GAPDoc Label="ListDirectory">
+##  <ManSection>
+##  <Func Name="ListDirectory" Arg="[dir]"/>
+##
+##  <Description>
+##  lists the files in directory <A>dir</A> (a string) 
+##  or the current directory if called with no arguments.
+##  </Description>
+##  </ManSection>
+##  <#/GAPDoc>
+DeclareGlobalFunction( "ListDirectory" );
+]]></Log>
+This is all put together in the file <F>example/makedocrel.g</F> which 
+builds the package documentation, calling 
+<Ref Func="MakeGAPDocDoc" BookName="gapdoc"/> with 
+locations of library files containing parts of the 
+documentation.
```

Comment:
Might also advertise AutoDoc here, where the equivalent of the above example is this:
```
#! @Arguments [dir]
#! @Description
#!  lists the files in directory <A>dir</A> (a string) 
#!  or the current directory if called with no arguments.
DeclareGlobalFunction( "ListDirectory" );
```